### ISSUE TITLE
Implementing efficient orthogonalization with QDWH

### ISSUE THREAD

**Issue by valentyn1boreiko** [NVIDIA-NeMo/Emerging-Optimizers#45]
**Is your feature request related to a problem? Please describe.**
QR-based Dynamically Weighted Halley (QDWH) can be used for orthogonalization and has cubic convergence (https://arxiv.org/abs/2505.21799). Would it be possible to add it as well as an alternative to Newton-Schulz?

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by skyw** [comment]
implemented a version via SVD. It is not too slow, should be enough for evaluation purpose and motivate kernel work if turned out to be promising.

**Comment by timlautk** [comment]
> PolarGrad

It might be a bit late, but as the author of PolarGrad, I would like to mention that we can also compute the nuclear norm term via Newton--Schulz or Polar Express. The nuclear norm term can be computed via torch.sum(grad, orth_grad) regardless of how orth_grad is computed; see the paper for why this is the formula. 

QDWH has the benefit of handling ill-conditioned gradients (could be beneficial to the embedding and head layers although people don't really use Muon or PolarGrad for these layers); see the latest version of [PolarGrad](https://arxiv.org/pdf/2505.21799) for detailed explanation on this.

**Comment by mkhona-nvidia** [comment]
@timlautk do you want to submit an MR to `OrthogonalizedOptimizer` and/or `Muon` in the codebase to optionally use polargrad's nuclear norm term? Or you can subclass `OrthogonalizedOptimizer`, overwrite the `orthogonalize` function and add in the nuclear norm term through `orth_grad@grad` so we can reuse newton-schulz

**Comment by skyw** [comment]
We decided to upstream to PyTorch, stay tuned. https://github.com/pytorch/pytorch/issues/175636

**Comment by skyw** [comment]
cusolver added QDWH in CUDA 13.2.1 release.

We'll pick it up once it is available in pytorch.
